lib/reporters/TestReporter.js

```diff
@@ -0,0 +1,92 @@
+import {Test, Suite} from "../Data.js";
+
+/*
+ The TestReporter verifies that a test runner outputs the right data in the right order.
+ To do so, it compares the actual output with the provided reference data.
+ The result is given in the ok attribute.
+ */
+export default class TestReporter {
+    /**
+     * @param runner: standardized test runner (or adapter)
+     * @param referenceData: An array of all expected (eventName, eventData) tuples in the right order
+     */
+    constructor(runner, referenceData) {
+        this.referenceData = referenceData.slice();
+        this.error = false;
+        runner.on("runStart", this.onEvent.bind(this, "runStart"));
+        runner.on("suiteStart", this.onEvent.bind(this, "suiteStart"));
+        runner.on("testStart", this.onEvent.bind(this, "testStart"));
+        runner.on("testEnd", this.onEvent.bind(this, "testEnd"));
+        runner.on("suiteEnd", this.onEvent.bind(this, "suiteEnd"));
+        runner.on("runEnd", this.onEvent.bind(this, "runEnd"));
+    }
+
+    /**
+     * Gets called on each event emitted by the runner. Checks if the actual event matches the expected event.
+     */
+    onEvent(eventName, eventData) {
+        var [expectedEventName, expectedEventData] = this.referenceData.shift();
+
+        if (expectedEventName !== eventName || !this.equal(eventData, expectedEventData)) {
```

Comment:
Tiny detail, but lets switch the order of the first comparison (eventName) to be consistent with the equal call.